benchmarks/src/main/java/io/scalecube/benchmarks/BenchmarksState.java

```diff
@@ -0,0 +1,188 @@
+  }
+
+  protected void afterAll() {
+    // NOP
+  }
+
+  /**
+   * Executes starting of the state, also it includes running of {@link BenchmarksState#beforeAll}.
+   */
+  public final void start() {
+    if (!started.compareAndSet(false, true)) {
+      throw new IllegalStateException("BenchmarksState is already started");
+    }
+
+    LOGGER.info("Benchmarks settings: " + settings);
+
+    consoleReporter = ConsoleReporter.forRegistry(settings.registry())
+        .outputTo(System.out)
+        .convertDurationsTo(settings.durationUnit())
+        .convertRatesTo(settings.rateUnit())
+        .build();
+
+    csvReporter = CsvReporter.forRegistry(settings.registry())
+        .convertDurationsTo(settings.durationUnit())
+        .convertRatesTo(settings.rateUnit())
+        .build(settings.csvReporterDirectory());
+
+    scheduler = Schedulers.fromExecutor(Executors.newFixedThreadPool(settings.nThreads()));
+
+    beforeAll();
+
+    consoleReporter.start(settings.reporterPeriod().toMillis(), TimeUnit.MILLISECONDS);
+    csvReporter.start(1, TimeUnit.DAYS);
+
+    Runtime.getRuntime().addShutdownHook(new Thread(() -> {
+      if (started.get()) {
+        csvReporter.report();
+        consoleReporter.report();
+      }
+    }));
+
+  }
+
+  /**
+   * Executes shutdown process of the state, also it includes running of {@link BenchmarksState#afterAll}.
+   */
+  public final void shutdown() {
+    if (!started.compareAndSet(true, false)) {
+      throw new IllegalStateException("BenchmarksState is not started");
+    }
+
+    if (consoleReporter != null) {
+      consoleReporter.report();
+      consoleReporter.stop();
+    }
+
+    if (csvReporter != null) {
+      csvReporter.report();
+      csvReporter.stop();
+    }
+
+    if (scheduler != null) {
+      scheduler.dispose();
+    }
+
+    afterAll();
+  }
+
+  public Scheduler scheduler() {
+    return scheduler;
+  }
+
+  /**
+   * Returns timer with specified name.
+   *
+   * @param name name
+   * @return timer with specified name
+   */
+  public Timer timer(String name) {
+    return settings.registry().timer(settings.taskName() + "-" + name);
+  }
+
+  /**
+   * Returns meter with specified name.
+   *
+   * @param name name
+   * @return meter with specified name
+   */
+  public Meter meter(String name) {
+    return settings.registry().meter(settings.taskName() + "-" + name);
+  }
+
+  /**
+   * Returns histogram with specified name.
+   *
+   * @param name name
+   * @return histogram with specified name
+   */
+  public Histogram histogram(String name) {
+    return settings.registry().histogram(settings.taskName() + "-" + name);
+  }
+
+  /**
+   * Runs given function in the state. It also executes {@link BenchmarksState#start()} before and
+   * {@link BenchmarksState#shutdown()} after.
+   * <p>
+   * NOTICE: It's only for synchronous code.
+   * </p>
+   *
+   * @param func function
+   */
+  public final void runForSync(Function<T, Function<Long, Object>> func) {
```

Comment:
consder using `LongFunction<Object>`
please document what do you expect *this* function to return.